torch/csrc/jit/fuser/cpu/fused_kernel.cpp

```diff
@@ -63,20 +189,38 @@ static CompilerConfig& getConfig() {
 // understand for AVX512. When we need better CPU performance this
 // optimization can be re-enabled by tracking down the platforms where
 // this error occurs and only selectively disabling it.
+#ifdef _MSC_VER
+static std::string getArchFlags() {
+  if (InstructionSet::AVX512F()) {
+    return "/arch:AVX512";
+  } else if (InstructionSet::AVX2()) {
+    return "/arch:AVX2";
+  } else if (InstructionSet::AVX()) {
+    return "/arch:AVX";
+  } else {
+    return "";
+  }
+}
+static const std::string arch_flags = getArchFlags();
+static const std::string compile_string =
+    "cd /D \"" + temp_dir + "\" && "
+    "${cxx} /nologo /MD /Ox " + arch_flags + " /LD /EHsc "
```

Comment:
Well, this library has nothing to do with the torch library because it doesn't include or link any sources/libraries from torch. I use /MD here only to keep consistent actually it doesn't rely on external dependencies so the runtime switch won't matter.